pkg/storage/filesystem/btrfs.go

```diff
@@ -591,6 +592,26 @@ func (v *btrfsVolume) Usage() (usage Usage, err error) {
 func (v *btrfsVolume) Limit(size uint64) error {
 	ctx := context.Background()
 
+	info, err := v.utils.SubvolumeInfo(ctx, v.Path())
+	if err != nil {
+		return err
+	}
+
+	groups, err := v.utils.QGroupList(ctx, v.Path())
+	if err != nil {
+		return err
+	}
+
+	// Check if there is a qgroup for before we try to limit
+	_, ok := groups[fmt.Sprintf("0/%d", info.ID)]
+	if !ok {
+		log.Debug().Msgf("Creating qgroup 0/%d, at %v", info.ID, v.Path())
+		err = v.utils.QGroupCreate(ctx, strconv.Itoa(info.ID), v.Path())
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
What is the exact reason for this logic being added? In the past there was just the QGroupLimited call, which took care of everything as far as I know. So what is the underlying reason that this might be needed?

Reply:
we've identify a node which failed to set the qgroup limit on the cache at the boot of storaged. Turns out the reason is there was no qgroup for this specific subvolume.

Reply:
@LeeSmet check the linked issue

Reply:
Previously we never explicitly created the qgroup. I understand that this works in the scenario you described, but I want to know why this is suddenly needed. I.E why can limit not implicitly create the subgroup anymore? And more importantly, why is the subgroup gone in the first place? That seems to indicate a bug somewhere else in the code to me, which we are just covering up here.

From the linked issue, the error is `ERROR: unable to limit requested quota group: No such file or directory\n`. Are we sure that is related to qgroups, or is there a possibility tha the path that was passed happened to just be wrong for some reason?

Reply:
It could indeed indicate that there is another issue somewhere else, perhaps maybe while testing the cleanup script and the script did not account for the zos-cache.. 

Also I debugged the error that was listed in the issue and it seemed like there was nothing wrong with passing the `path`. I also tried using btrfs command myself explicitly passing the path and it also failed. So that's how I verified this issue.

Reply:
> Also I debugged the error that was listed in the issue and it seemed like there was nothing wrong with passing the path. I also tried using btrfs command myself explicitly passing the path and it also failed

I'm confused, exactly what did you do and what succeeded/failed?

Reply:
> Are we sure that is related to qgroups, or is there a possibility tha the path that was passed happened to just be wrong for some reason?

I verified that this is not the issue

Reply:
so, if I understand correctly, the situation is as follows:

1. Cache is created
2. Cache quota is enabled en set to 100GB
3. Something happens, but we have no clue what
4. Storaged restarts
5. Storaged finds the mounted cache and tries to enable quota, but fails for an unknown reason

This suggests to me that there really is an issue somewhere else we should fix. Sure if we do this, it'll allow the cache mount flow to continue. But we still have no clue what really went wrong, and more importantly, what other issues this may cause that we just haven't noticed yet

Reply:
I double checked the cleanup routine and I did not find anything that could cause this issue again. I suggest we close this PR until we might see this issue pop up again so we can find the cause. I fixed this one node that had this issue.